gen2-rgbd-projection/projector_3d.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+
+# Code copied from main depthai repo, depthai_helpers/projector_3d.py
+
+import numpy as np
+import open3d as o3d
+
+class PointCloudVisualizer():
+    def __init__(self, intrinsic_matrix, width, height):
+        self.depth_map = None
+        self.rgb = None
+        self.pcl = None
+
+        self.pinhole_camera_intrinsic = o3d.camera.PinholeCameraIntrinsic(width,
+                                                                         height,
+                                                                         intrinsic_matrix[0][0],
+                                                                         intrinsic_matrix[1][1],
+                                                                         intrinsic_matrix[0][2],
+                                                                         intrinsic_matrix[1][2])
+        self.vis = o3d.visualization.Visualizer()
+        self.vis.create_window()
+        self.isstarted = False
+
+    def rgbd_to_projection(self, depth_map, rgb, is_rgb):
+        self.depth_map = depth_map
+        self.rgb = rgb
+        rgb_o3d = o3d.geometry.Image(self.rgb)
+        depth_o3d = o3d.geometry.Image(self.depth_map)
+        # TODO: query frame shape to get this, and remove the param 'is_rgb'
+        if is_rgb:
+            rgbd_image = o3d.geometry.RGBDImage.create_from_color_and_depth(rgb_o3d, depth_o3d, convert_rgb_to_intensity=False)
+        else:
+            rgbd_image = o3d.geometry.RGBDImage.create_from_color_and_depth(rgb_o3d, depth_o3d)
+        if self.pcl is None:
+            self.pcl = o3d.geometry.PointCloud.create_from_rgbd_image(rgbd_image, self.pinhole_camera_intrinsic)
+        else:
+            pcd = o3d.geometry.PointCloud.create_from_rgbd_image(rgbd_image, self.pinhole_camera_intrinsic)
+            self.pcl.points = pcd.points
+            self.pcl.colors = pcd.colors
+        return self.pcl
```

Comment:
There is lot of noise in your example than the old one. Can you compare and add some steps which cleans this ?
for example: `pcd.remove_non_finite_points()` Check Open3D docs for more